### PR TITLE
Mqtt/add config template

### PR DESCRIPTION
**Pull request by aggarw13**
**Goal of this PR** is to add a single file containing information of all MQTT config macros. 
Related Issue: #1066 

Changes in PR:
* Add a `mqtt_config_defaults.h` in `libraries/standard/mqtt/include` that acts as the aggregator of all config macros for convenient reference, and defines their default values. (All the default macro value logic is moved to this file).  
* Remove the `mqtt/src/cprivate/mqtt_internal.h` file to instead use the `mqtt_config_defaults.h` file for defining _away_ logging macros.
* (Unrelated to main purpose of PR) Move `MAX_SUBSCRIPTION_CALLBACK_RECORDS` config to `demo_config.h` file in connection sharing demo folder.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/include/mqtt_config.h.template [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * @brief Number of milliseconds to wait for a ping response to a ping
+ * request as part of the keep-alive mechanism.
+ *
+ * If a ping response is not received before this timeout, then
+ * #MQTT_ProcessLoop will return #MQTTKeepAliveTimeout.
+ *
+ *
+ * #define MQTT_PINGRESP_TIMEOUT_MS             ( 500U )
+ */
+
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/include/mqtt_config.h.template:77]: What about MQTT_MAX_CONNACK_RECEIVE_RETRY_COUNT?

**Comment by aggarw13** [comment]
> What about MQTT_MAX_CONNACK_RECEIVE_RETRY_COUNT? 

Thanks for pointing out. Added

File: libraries/standard/mqtt/include/mqtt_config.h.template (hunk 1 above)
Comment (aggarw13) [reply libraries/standard/mqtt/include/mqtt_config.h.template]: Added

File: libraries/standard/mqtt/include/mqtt_config.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+/**
+ * @brief The maximum number of MQTT PUBLISH messages that may be pending
+ * acknowledgement at any time.
+ *
+ * QoS 1 and 2 MQTT PUBLISHes require acknowledgement from the server before
+ * they can be completed. While they are awaiting the acknowledgement, the
+ * client must maintain information about their state. The value of this
+ * macro sets the limit on how many simultaneous PUBLISH states an MQTT
+ * context maintains.
```
Comment (leegeth) [inline libraries/standard/mqtt/include/mqtt_config.h:61]: I think we may be missing the information here that we maintain the incoming and outgoing publishes separately. Total number of Publishes including both incoming and outgoing publishes are 2* the config value.

Comment (aggarw13) [reply libraries/standard/mqtt/include/mqtt_config.h]: Good point, have updated the comment

**Comment by sarenameas** [comment]
Can you please run doxygen and fix your warnings.
```
cd libraries/standard/mqtt
doxygen doc/doxygen/config.doxyfiel
```

File: libraries/standard/mqtt/include/mqtt_config_defaults.h [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+/**
+ * @brief The maximum number of MQTT PUBLISH messages that may be pending
+ * acknowledgement at any time.
+ *
+ * QoS 1 and 2 MQTT PUBLISHes require acknowledgement from the server before
+ * they can be completed. While they are awaiting the acknowledgement, the
+ * client must maintain information about their state. The value of this
+ * macro sets the limit on how many simultaneous PUBLISH states an MQTT
+ * context maintains, separately, for both incoming and outgoing direction of
+ * PUBLISHes.
```
Comment (leegeth) [inline libraries/standard/mqtt/include/mqtt_config_defaults.h:44]: The @brief description and this statement may be conveying different meaning of this macro.

Comment (aggarw13) [reply libraries/standard/mqtt/include/mqtt_config_defaults.h:45]: Updated

File: libraries/standard/mqtt/include/mqtt_config_defaults.h [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+ * application-specific logging implementation that supports error logging.
+ *
+ * @note The logging macro is called in MQTT library with parameters wrapped in
+ * double parentheses to be ISO C89/C90 standard compliant. For an example
+ * implementation of the logging macros, refer to mqtt_config.h files of MQTT demo folder
+ * and the reference implementation of logging-stack in demos.
+ *
+ * <b>Default value</b>: Error logging is turned off, and no code is generated for calls
+ * to the macro in the MQTT library on compilation.
+ */
```
Comment (sarenameas) [inline libraries/standard/mqtt/include/mqtt_config_defaults.h:105]: Oh this is nice thanks

**Pull request merged**